nsv13/code/controllers/subsystem/starsystem.dm

```diff
@@ -297,16 +411,50 @@ Returns a faction datum by its name (case insensitive!)
 	var/preset_trader = null
 	var/datum/trader/trader = null
 	var/list/audio_cues = null //if you want music to queue on system entry. Format: list of youtube or media URLS.
-
 	var/already_announced_combat = FALSE
+	var/startup_proc = null
 
 /datum/star_system/proc/dist(datum/star_system/other)
 	var/dx = other.x - x
 	var/dy = other.y - y
 	return sqrt((dx * dx) + (dy * dy))
 
-/datum/star_system/New()
+/datum/star_system/proc/parse_startup_proc()
+	switch(startup_proc)
+		if("STARTUP_PROC_TYPE_BRASIL")
+			addtimer(CALLBACK(src, .proc/generate_badlands), 10 SECONDS)
+			return
+	message_admins("WARNING: Invalid startup_proc declared for [name]! Review your defines (~L438, starsystem.dm), please.")
+	return 1
+
+/datum/star_system/New(name, desc, threat_level, alignment, hidden, system_type, system_traits, is_capital, adjacency_list, wormhole_connections, fleet_type, x, y, parallax_property, visitable, sector, is_hypergate, preset_trader, audio_cues, startup_proc)
 	. = ..()
+	//Load props first.
+	if(name != null)
```

Comment:
I don't know if it's faster but it's a byond proc so I'd assume so 🤔 
```suggestion
	if(!isnull(name))
```

Reply:
I'm not sure what the byond jank is, but it's just checking if the pointer is 0 or not, I don't think we can really get much faster than that.

Language specifics may change this though

Reply:
Given that variables can store any type of data, I would _assume_ an evaluation operator checking for null would be slower than a proc just made to check for null pointers. However, I don't know a lot about byond's compiler optimizations so it could just scrap it's data type checking for null checks for all we know.

Either way isnull() looks nicer but this is so pedantic that it doesn't matter :)

Reply:
Can this entire check not just be if(name) though? 🤔

Reply:
Yeah true say, just habits from work